packages/mc-html-template/lib/compile-html.js

```diff
@@ -0,0 +1,66 @@
+const fs = require('fs');
+const path = require('path');
+const fetch = require('node-fetch');
+const loadEnv = require('./load-env');
+const loadHeaders = require('./load-headers');
+const replaceHtmlPlaceholders = require('./utils/replace-html-placeholders');
+
+const requiredOptions = ['envPath', 'publicAssetsPath'];
+
+/**
+ * Options:
+ * - envPath
+ * - cspPath
+ * - publicAssetsPath
+ * - useLocalAssets
+ */
+module.exports = async function compileHtml(options) {
+  requiredOptions.forEach(key => {
+    if (!options[key]) {
+      throw new Error(`Missing required option ${key}.`);
+    }
+  });
+
+  const env = loadEnv(options.envPath);
+  const headers = loadHeaders(env, { cspPath: options.cspPath });
+
+  // For local usage only!
+  if (options.useLocalAssets) {
+    const indexHtmlContent = fs.readFileSync(
+      path.join(options.publicAssetsPath, 'index.html.template'),
+      'utf8'
+    );
+    const updatedIndexHtmlContent = replaceHtmlPlaceholders(
+      indexHtmlContent,
+      env
+    );
+    return {
+      env,
+      headers,
+      indexHtmlContent: updatedIndexHtmlContent,
+    };
+  }
+
+  // TL;DR; This disables the `Cache-Control` and ensures that we get
+  // the latest version of the object.
+  const randomQueryParam = Date.now();
+  // Fetch `index.html.template` from remote CDN
+  const remoteIndexHtmlResponse = await fetch(
+    `${env.cdnUrl}/index.html.template?${randomQueryParam}`
```

Comment:
can you explain why we need to do this?

Reply:
What exactly? The "random query" part? It's to ensure that we don't get a cached version of the file, always the up-to-date version.

Reply:
I mean querying the cdn to get the template?

Reply:
Why can't  we have the template locally?

Reply:
That's how it works currently (this logic was part of the mc-http-server previously).

When we bundle each app, the js/css assets are generated by webpack as well as the `index.html.template`. We upload them to our CDN, then we deploy the http server, which on startup will fetch and compile the `index.html.template` for the specific app, thus serving the latest assets (referenced in the template by webpack).

Does it clarify your question?

Reply:
> Why can't we have the template locally?

You can optionally choose to have the template locally (together with the other assets) by passing the option `--use-local-assets`. That's not the recommended approach for production though.